3D Chess/Assets/Scripts/StartGameButton.cs

```diff
@@ -0,0 +1,64 @@
+using Realms;
+using Realms.Sync;
+using Realms.Sync.Exceptions;
+using System;
+using System.IO;
+using System.Threading.Tasks;
+using UnityEngine;
+using UnityEngine.SceneManagement;
+using UnityEngine.UI;
+
+public class StartGameButton : MonoBehaviour
+{
+    [SerializeField] private GameObject loadingIndicator = default;
+    [SerializeField] private InputField gameIdInputField = default;
+
+    public async void OnStartButtonClicked()
+    {
+        loadingIndicator.SetActive(true);
+
+        var gameId = gameIdInputField.text;
+        PlayerPrefs.SetString(Constants.PlayerPrefsKeys.GameId, gameId);
+
+        await CreateRealmAsync(gameId);
+
+        SceneManager.LoadScene(Constants.SceneNames.Main);
+    }
+
+    private async Task CreateRealmAsync(string gameId)
+    {
+        var app = App.Create(Constants.Realm.AppId);
+        var user = app.CurrentUser;
+
+        if (user == null)
+        {
+            // This example focuses on an introduction to Sync.
+            // We will keep the registration simple for now by just creating a random email and password.
+            // We'll also not create a seperate registration dialogue here and instead just try to register
+            // and if already registered, log in.
+            // In a different example we will focus on authentication methods, login / registration dialogues, etc.
+            var email = Guid.NewGuid().ToString();
+            var password = Guid.NewGuid().ToString();
+            try
+            {
+                await app.EmailPasswordAuth.RegisterUserAsync(email, password);
+            }
+            catch (AppException appException)
+            {
+                // If the user is already registered we can just log in. Any other exception needs to be thrown.
+                if (!appException.Message.Contains("AccountNameInUse"))
```

Comment:
For all intents and purposes, there's no way this will happen. Someone needs to generate a gazillion guids to have a chance of collision and they'll hit other limitations first (I'm fairly certain the server has a lower limit of users it can store).

Reply:
Fair point. That's a relict from having the name on the welcome screen and therefore then registering a proper name, not a random one. I'll remove it.